File: 1erdepartamental/python/8.promedio/core/estadisticas.py
# Estadísticas simples con 5 números fijos

#Clase Principal
class Estadisticas:
    def __init__(self, n1, n2, n3, n4, n5):
        self.__numeros = [n1, n2, n3, n4, n5] #Atributos guardados en lista
    
    def suma(self):
        return sum(self.__numeros)#suma todos los atributos que estan en la lista

    def promedio(self):#suma y divide todos los atributos que estan en la lista
        return self.suma() / len(self.__numeros)#len : numero de elementos en la lista

    def maximo(self):#maximo todos los atributos que estan en la lista
        return max(self.__numeros)

    def minimo(self):#minimo todos los atributos que estan en la lista
        return min(self.__numeros)

File: 1erdepartamental/c++/8.promedio/core/Estadisticas.cpp
#include "Estadisticas.h"

// Constructor - inicializa el arreglo
Estadisticas::Estadisticas(double n1, double n2, double n3, double n4, double n5) {
    __numeros = {n1, n2, n3, n4, n5};
}

// Obtiene el arreglo de numeros
std::array<double, 5> Estadisticas::getNumeros() const {
    return __numeros;
}

// Calcula la suma de todos los numeros
double Estadisticas::suma() const {
    double total = 0;
    for (double n : __numeros) total += n;
    return total;
}

// Calcula el promedio
double Estadisticas::promedio() const {
    return suma() / __numeros.size();
}

// Encuentra el valor maximo
double Estadisticas::maximo() const {
    double m = __numeros[0];
    for (double n : __numeros) {
        if (n > m) m = n;
    }
    return m;
}

// Encuentra el valor minimo
double Estadisticas::minimo() const {
    double m = __numeros[0];
    for (double n : __numeros) {
        if (n < m) m = n;
    }
    return m;
}
